xapian-letor/m4-macros/xapianletor.m4

```diff
@@ -52,56 +52,64 @@ AC_DEFUN([XO_LIB_XAPIANLETOR],
     ;;
   esac
   AC_PATH_PROG(XAPIANLETOR_CONFIG, "$config_script_to_check_for")
-  AC_MSG_CHECKING([$XAPIANLETOR_CONFIG works])
-  dnl check for --ltlibs but not --libs as "xapianletor-config --libs" will
-  dnl fail if xapian isn't installed...
-
-  dnl run with exec to avoid leaking output on "real" bourne shells
-  if (exec >&5 2>&5 ; $XAPIANLETOR_CONFIG --ltlibs --cxxflags; exit $?) then
-    AC_MSG_RESULT(yes)
+  if test -z "$XAPIANLETOR_CONFIG"; then
+    ifelse([$2], ,
+      [ifelse([$1], , [
+	AC_MSG_ERROR([Can't find xapianletor-config.  If you have installed the Xapian-letor library, you need to add XAPIANLETOR_CONFIG=/path/to/xapianletor-config to your configure command.])],
+	:)],
+      [$2])
   else
-    case $? in
-    127)
-    AC_MSG_ERROR(['$XAPIANLETOR_CONFIG' not found, aborting])
-    ;;
-    126)
-    if test -d "$XAPIANLETOR_CONFIG" ; then
-      AC_MSG_ERROR(['$XAPIANLETOR_CONFIG' is a directory; it should be the filename of the xapianletor-config script])
-    fi
-    AC_MSG_ERROR(['$XAPIANLETOR_CONFIG' not executable, aborting])
-    ;;
-    esac
+    AC_MSG_CHECKING([$XAPIANLETOR_CONFIG works])
+    dnl check for --ltlibs but not --libs as "xapianletor-config --libs" will
+    dnl continue if xapian-letor isn't installed...
```

Comment:
In `xapian.m4` the equivalent comment says `... will **fail** if xapian isn't installed".

I'm not sure why you've changed from "fail" to "continue" here - testing in an uninstalled source tree shows it fails for letor just like it does for core:

```
olly@gemse:~/git/xapian/xapian-letor$ ./xapianletor-config --libs
./xapianletor-config: Use libtool and 'xapianletor-config --ltlibs'
to link with an uninstalled libxapianletor.
olly@gemse:~/git/xapian/xapian-letor$ echo $?
1
```

Am I missing something?

Reply:
oh sorry, I don't know why I did it. fixed in a fixup commit.